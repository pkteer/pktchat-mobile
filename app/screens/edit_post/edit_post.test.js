// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {Keyboard} from 'react-native';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import EditPost from './edit_post';

describe('EditPost', () => {
    const baseProps = {
        actions: {
            editPost: jest.fn(),
        },
        post: {},
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <EditPost {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should dismiss keyboard on close', () => {
        Keyboard.dismiss = jest.fn();
        expect(Keyboard.dismiss).not.toHaveBeenCalled();

        const wrapper = shallowWithIntl(
            <EditPost {...baseProps}/>,
        );
        wrapper.instance().close();
        expect(Keyboard.dismiss).toHaveBeenCalledTimes(1);
    });

    test('should handle edit post action error', async () => {
        const error = 'error';
        const props = {
            ...baseProps,
            actions: {
                editPost: jest.fn().mockResolvedValueOnce({error}),
            },
        };

        const wrapper = shallowWithIntl(
            <EditPost {...props}/>,
        );
        const instance = wrapper.instance();
        instance.emitEditing = jest.fn();
        instance.setState = jest.fn();

        await instance.onEditPost();

        expect(instance.emitEditing.mock.calls).toEqual([
            [true],
            [false],
        ]);
        expect(instance.setState.mock.calls).toEqual([
            [{editing: true, error: null, errorExtra: null}],
            [{editing: false, error}, instance.focus],
        ]);
    });

    test('should handle edit post action success', async () => {
        const props = {
            ...baseProps,
            actions: {
                editPost: jest.fn().mockResolvedValueOnce({}),
            },
        };

        const wrapper = shallowWithIntl(
            <EditPost {...props}/>,
        );
        const instance = wrapper.instance();
        instance.emitEditing = jest.fn();
        instance.setState = jest.fn();

        await instance.onEditPost();

        expect(instance.emitEditing.mock.calls).toEqual([
            [true],
            [false],
        ]);
        expect(instance.setState.mock.calls).toEqual([
            [{editing: true, error: null, errorExtra: null}],
            [{editing: false}, instance.close],
        ]);
    });
});
