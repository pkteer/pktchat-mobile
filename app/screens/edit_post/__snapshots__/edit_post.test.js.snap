// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPost should match snapshot 1`] = `
<React.Fragment>
  <RNCSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
    testID="edit_post.screen"
  >
    <Connect(StatusBar) />
    <View
      style={
        Object {
          "backgroundColor": "rgba(63,67,80,0.03)",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "rgba(63,67,80,0.1)",
              "borderBottomWidth": 1,
              "borderTopColor": "rgba(63,67,80,0.1)",
              "borderTopWidth": 1,
              "marginTop": 2,
            },
            Object {
              "height": NaN,
            },
          ]
        }
      >
        <TextInputWithLocalizedPlaceholder
          allowFontScaling={true}
          blurOnSubmit={false}
          disableFullscreenUI={true}
          keyboardAppearance="light"
          keyboardType="default"
          multiline={true}
          numberOfLines={10}
          onChangeText={[Function]}
          onSelectionChange={[Function]}
          placeholder={
            Object {
              "defaultMessage": "Edit the post...",
              "id": "edit_post.editPost",
            }
          }
          placeholderTextColor="rgba(63,67,80,0.4)"
          style={
            Array [
              Object {
                "color": "#3f4350",
                "fontSize": 14,
                "padding": 15,
                "textAlignVertical": "top",
              },
              Object {
                "height": NaN,
              },
            ]
          }
          testID="edit_post.message.input"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </RNCSafeAreaView>
  <KeyboardTrackingView
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        },
        Object {
          "flex": 0,
        },
      ]
    }
  >
    <Connect(Autocomplete)
      cursorPosition={0}
      maxHeight={200}
      nestedScrollEnabled={true}
      offsetY={8}
      onChangeText={[Function]}
      onVisible={[Function]}
      style={
        Object {
          "position": undefined,
        }
      }
    />
  </KeyboardTrackingView>
</React.Fragment>
`;
